plugins/ltac/rewrite.ml

```diff
@@ -1892,14 +1892,11 @@ let declare_projection n instance_id r =
     let ctx,ccl = Reductionops.splay_prod_n env sigma (3 * n) typ
     in it_mkProd_or_LetIn ccl ctx
   in
-  let typ = it_mkProd_or_LetIn typ ctx in
-  let univs = Evd.univ_entry ~poly sigma in
-  let typ = EConstr.to_constr sigma typ in
-  let term = EConstr.to_constr sigma term in
-  let cst = Declare.definition_entry ~types:typ ~univs term in
-  let _ : Constant.t =
-    Declare.declare_constant ~name:n ~kind:Decls.(IsDefinition Definition)
-      (Declare.DefinitionEntry cst)
+  let types = Some (it_mkProd_or_LetIn typ ctx) in
+  let kind, opaque, scope = Decls.(IsDefinition Definition), false, DeclareDef.Global Declare.ImportDefaultBehavior in
+  let impargs, udecl = [], UState.default_univ_decl in
+  let _r : GlobRef.t =
+    DeclareDef.declare_definition ~name ~scope ~kind ~opaque ~impargs ~udecl ~poly ~types ~body sigma
```

Comment:
Here there is another semantic change, the code now is doing universe restriction, etc... usually this is what we want; but if not we could add the proper mode to `declare_definition`